src/middlewared/middlewared/service/decorators.py

```diff
@@ -162,6 +162,9 @@ def scrub(self, pool_name):
     :param abortable: If `True` then the job can be aborted in the task manager UI. When the job is aborted,
         `asyncio.CancelledError` is raised inside the job method (meaning that only asynchronous job methods can be
         aborted). By default, jobs are not abortable.
+
+    :param read_roles: A list of roles that will allow a non-full-admin user to see this job in `core.get_jobs`
+        and download its logs even if the job was launched by another user or by the system.
```

Comment:
We should also mention impact on `core.job_wait` and that by default users are able to read their own jobs. E.g. try to convince developer not to populate unless necessary.